"""Converting decades_tcc to the standard."""

In [1]:
import pandas as pd
import numpy as np
import s3fs
import os
import io
import boto3

import s3fs
fs = s3fs.S3FileSystem(anon=False,key='###',secret='###')

from dotenv import load_dotenv
load_dotenv(verbose=True)

def aws_session(region_name='us-east-1'):
    return boto3.session.Session(aws_access_key_id=os.getenv('AWS_ACCESS_KEY_ID'), #looks for any .env file
                                aws_secret_access_key=os.getenv('AWS_ACCESS_KEY_SECRET'), #Has to be in same directory
                                region_name=region_name) #from above

def make_bucket(name, acl): 
    session = aws_session()
    s3_resource = session.resource('s3')
    return s3_resource.create_bucket(Bucket=name, ACL=acl)

def upload_file_to_bucket(bucket_name, file_path):
    session = aws_session()
    s3_resource = session.resource('s3')
    file_dir, file_name = os.path.split(file_path)

    bucket = s3_resource.Bucket(bucket_name)
    bucket.upload_file(
      Filename=file_path,
      Key=file_name,
      ExtraArgs={'ACL': 'public-read'}
    )

    s3_url = f"https://{bucket_name}.s3.amazonaws.com/{file_name}"
    return s3_url

def download_file_from_bucket(bucket_name, s3_key, dst_path):
    session = aws_session()
    s3_resource = session.resource('s3')
    bucket = s3_resource.Bucket(bucket_name)
    bucket.download_file(Key=s3_key, Filename=dst_path)

## download_file_from_bucket('music-demo-lyrics', 'lyrics_25k.csv', 'short_name.csv')
## with open('short_name.csv') as fo:
    ## print(fo.read())

In [4]:
!aws s3 ls

2021-04-06 21:03:54 music-demo-lyrics
2021-04-06 20:10:43 music-lyrics
2021-04-07 07:23:53 worm-begin


In [6]:
!aws s3 ls 'worm-begin'

2021-04-07 08:38:33     105621 album_details_25k.csv
2021-04-07 08:09:22   27655251 decades_tcc_ceds_music.csv
2021-04-07 08:34:22     348120 genres_artists_data.csv
2021-04-07 08:19:38  276163416 genres_lyrics_data.csv
2021-04-07 08:22:31  187407184 labeled_lyrics_cleaned.csv
2021-04-07 08:38:18   40754845 lyrics_25k.csv
2021-04-07 08:38:34    2276377 songs_details_25k.csv


In [7]:
decades_df = pd.read_csv('s3://worm-begin/decades_tcc_ceds_music.csv')

In [8]:
decades_df

Unnamed: 0           artist_name                     track_name  \
0               0                mukesh           mohabbat bhi jhoothi   
1               4         frankie laine                      i believe   
2               6           johnnie ray                            cry   
3              10           pérez prado                       patricia   
4              12  giorgos papadopoulos             apopse eida oneiro   
...           ...                   ...                            ...   
28367       82447               mack 10                10 million ways   
28368       82448                m.o.p.  ante up (robbin hoodz theory)   
28369       82449                  nine                  whutcha want?   
28370       82450            will smith                         switch   
28371       82451                 jeezy                         r.i.p.   

       release_date    genre  \
0              1950      pop   
1              1950      pop   
2              1950      pop   
3              1950      pop   
4              1950      pop   
...             ...      ...   
28367          2019  hip hop   
28368          2019  hip hop   
28369          2019  hip hop   
28370          2019  hip hop   
28371          2019  hip hop   

                                                  lyrics  len    dating  \
0      hold time feel break feel untrue convince spea...   95  0.000598   
1      believe drop rain fall grow believe darkest ni...   51  0.035537   
2      sweetheart send letter goodbye secret feel bet...   24  0.002770   
3      kiss lips want stroll charm mambo chacha merin...   54  0.048249   
4      till darling till matter know till dream live ...   48  0.001350   
...                                                  ...  ...       ...   
28367  cause fuck leave scar tick tock clock come kno...   78  0.001350   
28368  minks things chain ring braclets yap fame come...   67  0.001284   
28369  get ban get ban stick crack relax plan attack ...   77  0.001504   
28370  check check yeah yeah hear thing call switch g...   67  0.001196   
28371  remix killer alive remix thriller trap bitch s...   83  0.001012   

       violence  world/life  ...   sadness  feelings  danceability  loudness  \
0      0.063746    0.000598  ...  0.380299  0.117175      0.357739  0.454119   
1      0.096777    0.443435  ...  0.001284  0.001284      0.331745  0.647540   
2      0.002770    0.002770  ...  0.002770  0.225422      0.456298  0.585288   
3      0.001548    0.001548  ...  0.225889  0.001548      0.686992  0.744404   
4      0.001350    0.417772  ...  0.068800  0.001350      0.291671  0.646489   
...         ...         ...  ...       ...       ...           ...       ...   
28367  0.001350    0.001350  ...  0.065664  0.001350      0.889527  0.759711   
28368  0.001284    0.035338  ...  0.001284  0.001284      0.662082  0.789580   
28369  0.154302    0.168988  ...  0.001504  0.001504      0.663165  0.726970   
28370  0.001196    0.001196  ...  0.001196  0.001196      0.883028  0.786888   
28371  0.075202    0.001012  ...  0.001012  0.033995      0.828875  0.674794   

       acousticness  instrumentalness   valence    energy       topic  \
0          0.997992          0.901822  0.339448  0.137110     sadness   
1          0.954819          0.000002  0.325021  0.263240  world/life   
2          0.840361          0.000000  0.351814  0.139112       music   
3          0.083935          0.199393  0.775350  0.743736    romantic   
4          0.975904          0.000246  0.597073  0.394375    romantic   
...             ...               ...       ...       ...         ...   
28367      0.062549          0.000000  0.751649  0.695686     obscene   
28368      0.004607          0.000002  0.922712  0.797791     obscene   
28369      0.104417          0.000001  0.838211  0.767761     obscene   
28370      0.007027          0.000503  0.508450  0.885882     obscene   
28371      0.015862          0.000000  0.475474  0.492477     obscen

In [9]:
decades_df.describe(include='all')

Unnamed: 0  artist_name track_name  release_date  genre  \
count   28372.000000        28372      28372  28372.000000  28372   
unique           NaN         5426      23689           NaN      7   
top              NaN  johnny cash    tonight           NaN    pop   
freq             NaN          190         17           NaN   7042   
mean    42946.323558          NaN        NaN   1990.236888    NaN   
std     24749.325492          NaN        NaN     18.487463    NaN   
min         0.000000          NaN        NaN   1950.000000    NaN   
25%     20391.250000          NaN        NaN   1975.000000    NaN   
50%     45405.500000          NaN        NaN   1991.000000    NaN   
75%     64090.500000          NaN        NaN   2007.000000    NaN   
max     82451.000000          NaN        NaN   2019.000000    NaN   

                                                   lyrics           len  \
count                                               28372  28372.000000   
unique                                              28372           NaN   
top     shall tell start friends yippee mesa dupree do...           NaN   
freq                                                    1           NaN   
mean                                                  NaN     73.028444   
std                                                   NaN     41.829831   
min                                                   NaN      1.000000   
25%                                                   NaN     42.000000   
50%                                                   NaN     63.000000   
75%                                                   NaN     93.000000   
max                                                   NaN    199.000000   

              dating      violence    world/life  ...       sadness  \
count   28372.000000  28372.000000  28372.000000  ...  28372.000000   
unique           NaN           NaN           NaN  ...           NaN   
top              NaN           NaN           NaN  ...           NaN   
freq             NaN           NaN           NaN  ...           NaN   
mean        0.021112      0.118396      0.120973  ...      0.129389   
std         0.052370      0.178684      0.172200  ...      0.181143   
min         0.000291      0.000284      0.000291  ...      0.000284   
25%         0.000923      0.001120      0.001170  ...      0.001144   
50%         0.001462      0.002506      0.006579  ...      0.005263   
75%         0.004049      0.192608      0.197793  ...      0.235113   
max         0.647706      0.981781      0.962105  ...      0.981424   

            feelings  danceability      loudness  acousticness  \
count   28372.000000  28372.000000  28372.000000  2.837200e+04   
unique           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN   
mean        0.030996      0.533348      0.665249  3.392347e-01   
std         0.071652      0.173218      0.108434  3.267143e-01   
min         0.000289      0.005415      0.000000  2.811248e-07   
25%         0.000993      0.412975      0.595364  3.423598e-02   
50%         0.001754      0.538612      0.679050  2.259028e-01   
75%         0.032622      0.656666      0.749026  6.325298e-01   
max         0.958810      0.993502      1.000000  1.000000e+00   

        instrumentalness       valence        energy    topic           age  
count       28372.000000  28372.000000  28372.000000    28372  28372.000000  
unique               NaN           NaN           NaN        8           NaN  
top                  NaN           NaN           NaN  sadness           NaN  
freq                 NaN           NaN           NaN     6096           NaN  
mean            0.080049      0.532864      0.569875      NaN      0.425187  
std             0.211245      0.250972      0.244385      NaN      0.264107  
min             0.000000      0.000000      0.000000      NaN      0.014286  
25%       

In [10]:
decades_df.keys()

Index(['Unnamed: 0', 'artist_name', 'track_name', 'release_date', 'genre',
       'lyrics', 'len', 'dating', 'violence', 'world/life', 'night/time',
       'shake the audience', 'family/gospel', 'romantic', 'communication',
       'obscene', 'music', 'movement/places', 'light/visual perceptions',
       'family/spiritual', 'like/girls', 'sadness', 'feelings', 'danceability',
       'loudness', 'acousticness', 'instrumentalness', 'valence', 'energy',
       'topic', 'age'],
      dtype='object')

In [17]:
decades2_df = pd.DataFrame((decades_df), columns=['original_csv','artist_name','track_name','link','lyrics','language','genre','release_date'])

In [18]:
decades2_df

original_csv           artist_name                     track_name  \
0               NaN                mukesh           mohabbat bhi jhoothi   
1               NaN         frankie laine                      i believe   
2               NaN           johnnie ray                            cry   
3               NaN           pérez prado                       patricia   
4               NaN  giorgos papadopoulos             apopse eida oneiro   
...             ...                   ...                            ...   
28367           NaN               mack 10                10 million ways   
28368           NaN                m.o.p.  ante up (robbin hoodz theory)   
28369           NaN                  nine                  whutcha want?   
28370           NaN            will smith                         switch   
28371           NaN                 jeezy                         r.i.p.   

       link                                             lyrics  language  \
0       NaN  hold time feel break feel untrue convince spea...       NaN   
1       NaN  believe drop rain fall grow believe darkest ni...       NaN   
2       NaN  sweetheart send letter goodbye secret feel bet...       NaN   
3       NaN  kiss lips want stroll charm mambo chacha merin...       NaN   
4       NaN  till darling till matter know till dream live ...       NaN   
...     ...                                                ...       ...   
28367   NaN  cause fuck leave scar tick tock clock come kno...       NaN   
28368   NaN  minks things chain ring braclets yap fame come...       NaN   
28369   NaN  get ban get ban stick crack relax plan attack ...       NaN   
28370   NaN  check check yeah yeah hear thing call switch g...       NaN   
28371   NaN  remix killer alive remix thriller trap bitch s...       NaN   

         genre  release_date  
0          pop          1950  
1          pop          1950  
2          pop          1950  
3          pop          1950  
4          pop          1950  
...        ...           ...  
28367  hip hop          2019  
28368  hip hop          2019  
28369  hip hop          2019  
28370  hip hop          2019  
28371  hip hop          2019  

[28372 rows x 8 columns]

In [19]:
decades2_df['original_csv'] = 'decades_tcc'

In [22]:
decades3_df = decades2_df.rename(columns={'track_name':'song_name','release_date':'date'})

In [23]:
decades3_df

original_csv           artist_name                      song_name  link  \
0      decades_tcc                mukesh           mohabbat bhi jhoothi   NaN   
1      decades_tcc         frankie laine                      i believe   NaN   
2      decades_tcc           johnnie ray                            cry   NaN   
3      decades_tcc           pérez prado                       patricia   NaN   
4      decades_tcc  giorgos papadopoulos             apopse eida oneiro   NaN   
...            ...                   ...                            ...   ...   
28367  decades_tcc               mack 10                10 million ways   NaN   
28368  decades_tcc                m.o.p.  ante up (robbin hoodz theory)   NaN   
28369  decades_tcc                  nine                  whutcha want?   NaN   
28370  decades_tcc            will smith                         switch   NaN   
28371  decades_tcc                 jeezy                         r.i.p.   NaN   

                                                  lyrics  language    genre  \
0      hold time feel break feel untrue convince spea...       NaN      pop   
1      believe drop rain fall grow believe darkest ni...       NaN      pop   
2      sweetheart send letter goodbye secret feel bet...       NaN      pop   
3      kiss lips want stroll charm mambo chacha merin...       NaN      pop   
4      till darling till matter know till dream live ...       NaN      pop   
...                                                  ...       ...      ...   
28367  cause fuck leave scar tick tock clock come kno...       NaN  hip hop   
28368  minks things chain ring braclets yap fame come...       NaN  hip hop   
28369  get ban get ban stick crack relax plan attack ...       NaN  hip hop   
28370  check check yeah yeah hear thing call switch g...       NaN  hip hop   
28371  remix killer alive remix thriller trap bitch s...       NaN  hip hop   

       date  
0      1950  
1      1950  
2      1950  
3      1950  
4      1950  
...     ...  
28367  2019  
28368  2019  
28369  2019  
28370  2019  
28371  2019  

[28372 rows x 8 columns]

In [24]:
decades3_df[decades3_df.duplicated()]

Empty DataFrame
Columns: [original_csv, artist_name, song_name, link, lyrics, language, genre, date]
Index: []

""" I see the date here in decades, and I'd like to add that back into the genres main csv.  The genres series has a csv with date (by artist, not song, unfortunately). But I need to strip the "/" s from the singer_name.  And take off 'Lyrics' from the end of the artist name in the genres csv with years in it."""

"""WARNING!: Dead end just ahead...but I did learn some regex."""

In [26]:
merged2_df = pd.read_csv('merged2_genre_df.csv')

In [38]:
merged2_df

original_csv       artist_name                            song_name  \
0        genres_csv   /10000-maniacs/                       More Than This   
1        genres_csv   /10000-maniacs/                       More Than This   
2        genres_csv   /10000-maniacs/                    Because The Night   
3        genres_csv   /10000-maniacs/                    Because The Night   
4        genres_csv   /10000-maniacs/                       These Are Days   
...             ...               ...                                  ...   
124225   genres_csv           /sambo/                              Valerie   
124226   genres_csv           /sambo/                           Wake Me Up   
124227   genres_csv           /sambo/                           Wake Me Up   
124228   genres_csv       /seu-jorge/                                Don't   
124229   genres_csv  /zeca-pagodinho/  Deixa A Vida Me Levar (translation)   

                                                     link  \
0                      /10000-maniacs/more-than-this.html   
1                      /10000-maniacs/more-than-this.html   
2                   /10000-maniacs/because-the-night.html   
3                   /10000-maniacs/because-the-night.html   
4                      /10000-maniacs/these-are-days.html   
...                                                   ...   
124225                                /sambo/valerie.html   
124226                             /sambo/wake-me-up.html   
124227                             /sambo/wake-me-up.html   
124228                               /seu-jorge/dont.html   
124229  /zeca-pagodinho/deixa-a-vida-me-levar-translat...   

                                                   lyrics language  genre  
0       I could feel at the time. There was no way of ...  ENGLISH   Rock  
1       I could feel at the time. There was no way of ...  ENGLISH    Pop  
2       Take me now, baby, here as I am. Hold me close...  ENGLISH   Rock  
3       Take me now, baby, here as I am. Hold me close...  ENGLISH    Pop  
4       These are. These are days you'll remember. Nev...  ENGLISH   Rock  
...                                                   ...      ...    ...  
124225  Well sometimes I go out by myself. And I look ...  ENGLISH   Rock  
124226  Feeling my way through the darkness. Guided by...  ENGLISH  Samba  
124227  Feeling my way through the darkness. Guided by...  ENGLISH   Rock  
124228  Don't, don't, that's what you say. Each time t...  ENGLISH  Samba  
124229  I've been. For almost everything in life. In t...  ENGLISH  Samba  

[124230 rows x 7 columns]

In [34]:
punct = "\n\r"+string.punctuation
merged2_df['artist_name'] = merged2_df['artist_name'].str.translate(str.maketrans('','',punct))

AttributeError: type object 'str' has no attribute 'punctuation'

In [39]:
merged3_df = merged2_df['artist_name'].str.replace(r'[^\w]', ' ')

<ipython-input-39-9eb4c0d0d6d4>:1: FutureWarning: The default value of regex will change from True to False in a future version.
  merged3_df = merged2_df['artist_name'].str.replace(r'[^\w]', ' ')


In [37]:
merged3_df

0          10000maniacs
1          10000maniacs
2          10000maniacs
3          10000maniacs
4          10000maniacs
              ...      
124225            sambo
124226            sambo
124227            sambo
124228         seujorge
124229    zecapagodinho
Name: artist_name, Length: 124230, dtype: object

In [28]:
download_file_from_bucket('worm-begin', 'album_details_25k.csv', 'year_df.csv')

In [30]:
year_df = pd.read_csv('year_df.csv')

In [40]:
year_df

Unnamed: 0       id          singer_name  \
0              0   5765.0  Taylor Swift Lyrics   
1              1   6432.0  Taylor Swift Lyrics   
2              2   6995.0  Taylor Swift Lyrics   
3              3  10358.0  Taylor Swift Lyrics   
4              4  24353.0  Taylor Swift Lyrics   
...          ...      ...                  ...   
1814        1814  22624.0    Jason Mraz Lyrics   
1815        1815  32642.0    Jason Mraz Lyrics   
1816        1816  58384.0    Jason Mraz Lyrics   
1817        1817  81696.0    Jason Mraz Lyrics   
1818        1818      NaN    Jason Mraz Lyrics   

                                                   name         type  \
0                                          Taylor Swift        album   
1     Sounds Of The Season: The Taylor Swift Holiday...           EP   
2                                              Fearless        album   
3                                             Speak Now        album   
4                                                   Red        album   
...                                                 ...          ...   
1814                         Love Is A Four Letter Word        album   
1815                                               Yes!        album   
1816                                              Know.        album   
1817                                  Look For The Good        album   
1818                                       other songs:  other songs   

             year  
0            2006  
1            2007  
2            2008  
3            2010  
4            2012  
...           ...  
1814         2012  
1815         2014  
1816         2018  
1817         2020  
1818  Not Defined  

[1819 rows x 6 columns]

In [32]:
pat = r'\b(?:{})\b'.format('|'.join(remove_words))

pat
'\\b(?:Lyrics)\\b'

year_df['singer_name'] = year_df['singer_name'].str.replace(pat, '')

NameError: name 'remove_words' is not defined

"""Trying some regex work now""" 

In [43]:
import numpy as np
year_df['test'] = year_df['singer_name'].str.extract(r'(^w{7})')
year_df

Unnamed: 0       id          singer_name  \
0              0   5765.0  Taylor Swift Lyrics   
1              1   6432.0  Taylor Swift Lyrics   
2              2   6995.0  Taylor Swift Lyrics   
3              3  10358.0  Taylor Swift Lyrics   
4              4  24353.0  Taylor Swift Lyrics   
...          ...      ...                  ...   
1814        1814  22624.0    Jason Mraz Lyrics   
1815        1815  32642.0    Jason Mraz Lyrics   
1816        1816  58384.0    Jason Mraz Lyrics   
1817        1817  81696.0    Jason Mraz Lyrics   
1818        1818      NaN    Jason Mraz Lyrics   

                                                   name         type  \
0                                          Taylor Swift        album   
1     Sounds Of The Season: The Taylor Swift Holiday...           EP   
2                                              Fearless        album   
3                                             Speak Now        album   
4                                                   Red        album   
...                                                 ...          ...   
1814                         Love Is A Four Letter Word        album   
1815                                               Yes!        album   
1816                                              Know.        album   
1817                                  Look For The Good        album   
1818                                       other songs:  other songs   

             year test  
0            2006  NaN  
1            2007  NaN  
2            2008  NaN  
3            2010  NaN  
4            2012  NaN  
...           ...  ...  
1814         2012  NaN  
1815         2014  NaN  
1816         2018  NaN  
1817         2020  NaN  
1818  Not Defined  NaN  

[1819 rows x 7 columns]

"""str.extract was a bust.  Is it not a str?"""

In [44]:
merged2_df.replace('(/)','',regex=True, inplace = True)
merged2_df

original_csv     artist_name                            song_name  \
0        genres_csv   10000-maniacs                       More Than This   
1        genres_csv   10000-maniacs                       More Than This   
2        genres_csv   10000-maniacs                    Because The Night   
3        genres_csv   10000-maniacs                    Because The Night   
4        genres_csv   10000-maniacs                       These Are Days   
...             ...             ...                                  ...   
124225   genres_csv           sambo                              Valerie   
124226   genres_csv           sambo                           Wake Me Up   
124227   genres_csv           sambo                           Wake Me Up   
124228   genres_csv       seu-jorge                                Don't   
124229   genres_csv  zeca-pagodinho  Deixa A Vida Me Levar (translation)   

                                                     link  \
0                        10000-maniacsmore-than-this.html   
1                        10000-maniacsmore-than-this.html   
2                     10000-maniacsbecause-the-night.html   
3                     10000-maniacsbecause-the-night.html   
4                        10000-maniacsthese-are-days.html   
...                                                   ...   
124225                                  sambovalerie.html   
124226                               sambowake-me-up.html   
124227                               sambowake-me-up.html   
124228                                 seu-jorgedont.html   
124229  zeca-pagodinhodeixa-a-vida-me-levar-translatio...   

                                                   lyrics language  genre  
0       I could feel at the time. There was no way of ...  ENGLISH   Rock  
1       I could feel at the time. There was no way of ...  ENGLISH    Pop  
2       Take me now, baby, here as I am. Hold me close...  ENGLISH   Rock  
3       Take me now, baby, here as I am. Hold me close...  ENGLISH    Pop  
4       These are. These are days you'll remember. Nev...  ENGLISH   Rock  
...                                                   ...      ...    ...  
124225  Well sometimes I go out by myself. And I look ...  ENGLISH   Rock  
124226  Feeling my way through the darkness. Guided by...  ENGLISH  Samba  
124227  Feeling my way through the darkness. Guided by...  ENGLISH   Rock  
124228  Don't, don't, that's what you say. Each time t...  ENGLISH  Samba  
124229  I've been. For almost everything in life. In t...  ENGLISH  Samba  

[124230 rows x 7 columns]

"""One down.  One to go."""
"""Tried a couple different .iloc and .loc combinations.  Trying to avoid making changes to a slice warnings.  Didn't work."""

In [47]:
year_df.loc['singer_name'].replace(('Lyrics'), '', regex = True, inplace = True)

KeyError: 'singer_name'

In [49]:
year_df['singer_name'].replace(('Lyrics'), '', regex = True, inplace = True)

In [50]:
year_df

Unnamed: 0       id    singer_name  \
0              0   5765.0  Taylor Swift    
1              1   6432.0  Taylor Swift    
2              2   6995.0  Taylor Swift    
3              3  10358.0  Taylor Swift    
4              4  24353.0  Taylor Swift    
...          ...      ...            ...   
1814        1814  22624.0    Jason Mraz    
1815        1815  32642.0    Jason Mraz    
1816        1816  58384.0    Jason Mraz    
1817        1817  81696.0    Jason Mraz    
1818        1818      NaN    Jason Mraz    

                                                   name         type  \
0                                          Taylor Swift        album   
1     Sounds Of The Season: The Taylor Swift Holiday...           EP   
2                                              Fearless        album   
3                                             Speak Now        album   
4                                                   Red        album   
...                                                 ...          ...   
1814                         Love Is A Four Letter Word        album   
1815                                               Yes!        album   
1816                                              Know.        album   
1817                                  Look For The Good        album   
1818                                       other songs:  other songs   

             year test  
0            2006  NaN  
1            2007  NaN  
2            2008  NaN  
3            2010  NaN  
4            2012  NaN  
...           ...  ...  
1814         2012  NaN  
1815         2014  NaN  
1816         2018  NaN  
1817         2020  NaN  
1818  Not Defined  NaN  

[1819 rows x 7 columns]

'''Yes!!"""

In [52]:
year2_df = pd.DataFrame((year_df), columns=['singer_name','year'])
year2_df

singer_name         year
0     Taylor Swift          2006
1     Taylor Swift          2007
2     Taylor Swift          2008
3     Taylor Swift          2010
4     Taylor Swift          2012
...             ...          ...
1814    Jason Mraz          2012
1815    Jason Mraz          2014
1816    Jason Mraz          2018
1817    Jason Mraz          2020
1818    Jason Mraz   Not Defined

[1819 rows x 2 columns]

In [54]:
year2_df.groupby(['singer_name']).count()

year
singer_name                      
'N Sync                         6
5 Seconds Of Summer (5SOS)     11
50 Cent                        20
6ix9ine (Tekashi69)             3
A Boogie Wit Da Hoodie          8
...                           ...
Twenty One Pilots               6
Westlife                       13
Wham!                           3
Whitney Houston                10
Yola                            2

[148 rows x 1 columns]

In [55]:
merged2_df.groupby(['artist_name']).count()

original_csv  song_name  link  lyrics  language  \
artist_name                                                                  
10000-maniacs                       264        264   264     264       264   
12-stones                           100        100   100     100       100   
2-chainz                            102        102   102     102       102   
21-savage                            11         11    11      11        11   
2ne1                                 48         48    48      48        48   
...                                 ...        ...   ...     ...       ...   
zeca-pagodinho                        1          1     1       1         1   
zelia-duncan                          3          3     3       3         3   
zendaya                              66         66    66      66        66   
zeze-di-camargo-e-luciano             5          5     5       5         5   
zz-top                              197        197   197     197       197   

                           genre  
artist_name                       
10000-maniacs                264  
12-stones                    100  
2-chainz                     102  
21-savage                     11  
2ne1                          48  
...                          ...  
zeca-pagodinho                 1  
zelia-duncan                   3  
zendaya                       66  
zeze-di-camargo-e-luciano      5  
zz-top                       197  

[1151 rows x 6 columns]

In [56]:
year2_df=year2_df.rename(columns={'singer_name':'artist_name'})
year2_df

artist_name         year
0     Taylor Swift          2006
1     Taylor Swift          2007
2     Taylor Swift          2008
3     Taylor Swift          2010
4     Taylor Swift          2012
...             ...          ...
1814    Jason Mraz          2012
1815    Jason Mraz          2014
1816    Jason Mraz          2018
1817    Jason Mraz          2020
1818    Jason Mraz   Not Defined

[1819 rows x 2 columns]

"""so, there are 148 artists in year_df, and ~10 times that amount in merged_df.  At this point, I'm doing it for the training."""

In [57]:
merged2_df

original_csv     artist_name                            song_name  \
0        genres_csv   10000-maniacs                       More Than This   
1        genres_csv   10000-maniacs                       More Than This   
2        genres_csv   10000-maniacs                    Because The Night   
3        genres_csv   10000-maniacs                    Because The Night   
4        genres_csv   10000-maniacs                       These Are Days   
...             ...             ...                                  ...   
124225   genres_csv           sambo                              Valerie   
124226   genres_csv           sambo                           Wake Me Up   
124227   genres_csv           sambo                           Wake Me Up   
124228   genres_csv       seu-jorge                                Don't   
124229   genres_csv  zeca-pagodinho  Deixa A Vida Me Levar (translation)   

                                                     link  \
0                        10000-maniacsmore-than-this.html   
1                        10000-maniacsmore-than-this.html   
2                     10000-maniacsbecause-the-night.html   
3                     10000-maniacsbecause-the-night.html   
4                        10000-maniacsthese-are-days.html   
...                                                   ...   
124225                                  sambovalerie.html   
124226                               sambowake-me-up.html   
124227                               sambowake-me-up.html   
124228                                 seu-jorgedont.html   
124229  zeca-pagodinhodeixa-a-vida-me-levar-translatio...   

                                                   lyrics language  genre  
0       I could feel at the time. There was no way of ...  ENGLISH   Rock  
1       I could feel at the time. There was no way of ...  ENGLISH    Pop  
2       Take me now, baby, here as I am. Hold me close...  ENGLISH   Rock  
3       Take me now, baby, here as I am. Hold me close...  ENGLISH    Pop  
4       These are. These are days you'll remember. Nev...  ENGLISH   Rock  
...                                                   ...      ...    ...  
124225  Well sometimes I go out by myself. And I look ...  ENGLISH   Rock  
124226  Feeling my way through the darkness. Guided by...  ENGLISH  Samba  
124227  Feeling my way through the darkness. Guided by...  ENGLISH   Rock  
124228  Don't, don't, that's what you say. Each time t...  ENGLISH  Samba  
124229  I've been. For almost everything in life. In t...  ENGLISH  Samba  

[124230 rows x 7 columns]

In [59]:
new_main_genres_df = pd.merge(merged2_df, year2_df, on='artist_name')
new_main_genres_df

Empty DataFrame
Columns: [original_csv, artist_name, song_name, link, lyrics, language, genre, year]
Index: []

In [60]:
year2_df[year2_df['artist_name'].str.match('Jason Mraz')== True]

artist_name         year
1808  Jason Mraz          2002
1809  Jason Mraz          2002
1810  Jason Mraz          2004
1811  Jason Mraz          2005
1812  Jason Mraz          2008
1813  Jason Mraz          2010
1814  Jason Mraz          2012
1815  Jason Mraz          2014
1816  Jason Mraz          2018
1817  Jason Mraz          2020
1818  Jason Mraz   Not Defined

In [66]:
merged2_df[merged2_df['artist_name'].str.match('Johnny Cash')== True]

Empty DataFrame
Columns: [original_csv, artist_name, song_name, link, lyrics, language, genre]
Index: []

"""I have been trying to pd.merge two csv files from different datasets.  There wasn't a Union between 'artist_name' s"""

In [67]:
decades3_df

original_csv           artist_name                      song_name  link  \
0      decades_tcc                mukesh           mohabbat bhi jhoothi   NaN   
1      decades_tcc         frankie laine                      i believe   NaN   
2      decades_tcc           johnnie ray                            cry   NaN   
3      decades_tcc           pérez prado                       patricia   NaN   
4      decades_tcc  giorgos papadopoulos             apopse eida oneiro   NaN   
...            ...                   ...                            ...   ...   
28367  decades_tcc               mack 10                10 million ways   NaN   
28368  decades_tcc                m.o.p.  ante up (robbin hoodz theory)   NaN   
28369  decades_tcc                  nine                  whutcha want?   NaN   
28370  decades_tcc            will smith                         switch   NaN   
28371  decades_tcc                 jeezy                         r.i.p.   NaN   

                                                  lyrics  language    genre  \
0      hold time feel break feel untrue convince spea...       NaN      pop   
1      believe drop rain fall grow believe darkest ni...       NaN      pop   
2      sweetheart send letter goodbye secret feel bet...       NaN      pop   
3      kiss lips want stroll charm mambo chacha merin...       NaN      pop   
4      till darling till matter know till dream live ...       NaN      pop   
...                                                  ...       ...      ...   
28367  cause fuck leave scar tick tock clock come kno...       NaN  hip hop   
28368  minks things chain ring braclets yap fame come...       NaN  hip hop   
28369  get ban get ban stick crack relax plan attack ...       NaN  hip hop   
28370  check check yeah yeah hear thing call switch g...       NaN  hip hop   
28371  remix killer alive remix thriller trap bitch s...       NaN  hip hop   

       date  
0      1950  
1      1950  
2      1950  
3      1950  
4      1950  
...     ...  
28367  2019  
28368  2019  
28369  2019  
28370  2019  
28371  2019  

[28372 rows x 8 columns]

In [68]:
decades3_df.to_csv('decades3_df.csv', index= False)

In [69]:
upload_file_to_bucket('music-demo-lyrics','decades3_df.csv')

'https://music-demo-lyrics.s3.amazonaws.com/decades3_df.csv'